Eclipse che docs contribution (#60)
eclipse-che/blog: _posts/2024-08-09-contributing-to-eclipse-che-docs.adoc

---

> title: Contributing to Eclipse Che documentation

Reviewer: IMO we need a better title ;-)

Author: I'll figure out something catchier

Reviewer: up to you, if you would like to keep it works as well

you can do it all in the environment you're already familiar with: an Eclipse Che cloud development environment (CDE).

image::/assets/img/contributing-to-eclipse-che-docs/che-docs.png[An Eclipse Che CDE created from the che-docs repository.]

== Getting started

Click the Developer Workspace button in the che-docs repository README file.
This opens a clone of the repository in an Eclipse Che workspace hosted
by Red Hat in the link:https://developers.redhat.com/developer-sandbox[Developer Sandbox].
Alternatively, you can click link:https://workspaces.openshift.com#https://github.com/eclipse/che-docs[*right here*].

== How does it work?

What about the che-docs repository makes using an Eclipse Che CDE to contribute content so convenient? The following two elements:

* link:https://eclipse.dev/che/docs/stable/end-user-guide/devfile-introduction/[devfile]
* list of link:https://eclipse.dev/che/docs/stable/end-user-guide/microsoft-visual-studio-code-open-source-ide/#automating-installation-of-microsoft-visual-studio-code-extensions-at-workspace-startup[automatically installed Microsoft Visual Studio Code extensions]

Let's look at the devfile first.

> The che-docs repository contains a simple devfile

Reviewer: ```suggestion
The che-docs repository contains a simple [devfile](https://github.com/eclipse-che/che-docs/blob/main/devfile.yaml)
```

> The che-docs repository contains a simple devfile

Reviewer: ```suggestion
The che-docs repository contains a simple link:https://github.com/eclipse-che/che-docs/blob/main/devfile.yaml[devfile]
```

that defines the basic features of the che-docs workspace.
It also includes a command for building a preview of your newly created content:

[source, code]
----

> # sh tools/preview.sh

Reviewer: for some reason executing the command does not work for me 
<img width="1521" alt="Screenshot 2024-08-07 at 19 40 14" src="https://github.com/user-attachments/assets/38549f69-8142-41ef-a310-26431a2ce5b0">

Reviewer: @dkwon17 @deerskindoll does it work for you?

Author: oh no. same result. I tried to run the script locally and there's an error message: `tools/preview.sh: line 15: gulp: command not found`

Reviewer: locally you might not have gulp ;-) It should work on Developer Sandbox OOTB because all dependencies are part of the image - that's exactly the problem we are solving - contributing with zero install

Author: my point is that it worked locally without issues. I'll look into it.

Reviewer: once fixed I believe we need to update the first screenshot with the workspace & preview image

Author: it works again. please check if it works for you, too.

Reviewer: works like a charm \o/ 
please, replace the first screenshot with:
<img width="1727" alt="Screenshot 2024-08-09 at 11 37 51" src="https://github.com/user-attachments/assets/8e363c05-620d-4786-89c4-bffdfa017b95">

----

This is helpful not only for checking the layout of the document but also for making sure that embedded images,
links and code blocks are displayed correctly.

By default, the che-docs workspace launches with Visual Studio Code - Open Source ("Code - OSS") as the IDE.
This is relevant for the second file mentioned above.

> The che-docs repository features an `extensions.json` file listing Visual Studio Code extensions

Reviewer: ```suggestion
The che-docs repository features an link:https://github.com/eclipse-che/che-docs/blob/main/.vscode/extensions.json[extensions.json] file listing Visual Studio Code extensions
```

you need for writing Eclipse Che content.
When you launch the che-docs workspace for the first time,

> these extensions are installed automatically.

Reviewer: @deerskindoll could you please add a screenshot here ?
<img width="347" alt="Screenshot 2024-08-07 at 19 24 54" src="https://github.com/user-attachments/assets/bc34d629-5be4-4844-bab3-7427ae2ff4f3">

Author: done

You can start working immediately,
with minimal additional setup required.

== Getting started

By launching the che-docs workspace, you're almost ready to go:

* You have a clone of the che-docs repository
* You have an IDE to write in.
* You have required extensions installed.
* You have a preview build script.

There are only three things left to do.

. Read the link:https://github.com/eclipse-che/che-docs/blob/main/CONTRIBUTING.adoc[contribution guidelines].
. Read how to link:https://redhat-documentation.github.io/supplementary-style-guide/#technical-examples[document commands] for Eclipse Che documentation.
. Update the grammar- and spelling-checking link:https://marketplace.visualstudio.com/items?itemName=ChrisChinchilla.vale-vscode[Vale extension] by running the following command in the terminal:
+
[source, code]
----
# vale sync
----

> 

Reviewer: I think we should define this command on devfile level

Let's look at the devfile first.
The che-docs repository contains a simple link:https://github.com/eclipse-che/che-docs/blob/main/devfile.yaml[devfile]
that defines the basic features of the che-docs workspace.
It also includes a command for building a preview of your newly created content:

[source, code]
----
# sh tools/preview.sh
----

This is helpful not only for checking the layout of the document but also for making sure that embedded images,
links and code blocks are displayed correctly.

By default, the che-docs workspace launches with Visual Studio Code - Open Source ("Code - OSS") as the IDE.
This is relevant for the second file mentioned above.
The che-docs repository features an link:https://github.com/eclipse-che/che-docs/blob/main/.vscode/extensions.json[extensions.json] file listing Visual Studio Code extensions
you need for writing Eclipse Che content.
When you launch the che-docs workspace for the first time,

> these extensions are installed automatically.

Reviewer: ```suggestion
these extensions are installed automatically.

image::/assets/img/contributing-to-eclipse-che-docs/extensions.png[Overview of automatically installed extensions.]

```

You can start working immediately,
with minimal additional setup required.

> 
> image::/assets/img/contributing-to-eclipse-che-docs/extensions.png[Overview of automatically installed extensions.]

Reviewer: ```suggestion
```

Reviewer: Moving image a few lines above ^

== Getting started

By launching the che-docs workspace, you're almost ready to go:

* You have a clone of the che-docs repository
* You have an IDE to write in.
* You have required extensions installed.
* You have a preview build script.

There are only three things left to do.

. Read the link:https://github.com/eclipse-che/che-docs/blob/main/CONTRIBUTING.adoc[contribution guidelines].
. Read how to link:https://redhat-documentation.github.io/supplementary-style-guide/#technical-examples[document commands] for Eclipse Che documentation.
. Update the grammar- and spelling-checking link:https://marketplace.visualstudio.com/items?itemName=ChrisChinchilla.vale-vscode[Vale extension] by running the following command in the terminal:
+
[source, code]
----

> # vale sync

Reviewer: it looks like it is part of the https://github.com/eclipse-che/che-docs/blob/main/tools/preview.sh#L14
so this section with vale should probably be described with preview.sh

Author: it's still better to run it separately, because this way, errors get flagged immediately, before you run the preview script

Reviewer: ok, but it should be explained in this case

Reviewer: also, if running this command separately makes sense please contribute it to the devfile directly

Author: that's why I specifically mentioned it as a prerequisite :D should I add an explanation that the preview script syncs vale as well?

Reviewer: if you want to leave it in this place let's add TIP / NOTE below:
`TIP: vale sync is part of devfile preview command, but it is recommended to run it that way because this and that`

Reviewer: PR for adding `vale-sync` command to devfile - https://github.com/eclipse-che/che-docs/pull/2768

---
title: Writing documentation in Eclipse Che
layout: post
author: Jana Vrbkova
description: >-
  Write documentation using an Eclipse Che CDE.
categories: []
keywords: ['che-docs', 'documentation', 'write', 'contribute']
slug: /@deerskindoll/writing-docs-with-che
---

== The quick and easy way to create documentation

With Eclipse Che, writing technical documentation is even easier than you think.
Let's use the link:https://github.com/eclipse-che/che-docs[Eclipse Che documentation repository on GitHub]
The che-docs repository was created with community contribution in mind,
featuring tools you need to create and test your content.
And from the initial draft to submitting a PR,
everything is happening in the environment you're already familiar with: an Eclipse Che cloud development environment (CDE).

image::/assets/img/contributing-to-eclipse-che-docs/workspace-with-preview.png[An Eclipse Che CDE created from the che-docs repository.]

== Getting started

> Click the Developer Workspace button in the che-docs repository README file.

Reviewer: ```suggestion
Click the Developer Workspace button in the che-docs repository link:https://github.com/eclipse-che/che-docs?tab=readme-ov-file#eclipse-che-documentation-project[README] file.
```

This opens a clone of the repository in an Eclipse Che workspace hosted
by Red Hat in the link:https://developers.redhat.com/developer-sandbox[Developer Sandbox].
Alternatively, you can click right here:

> [link=https://workspaces.openshift.com/f?url=https://github.com/eclipse/che-docs]

Reviewer: ```suggestion
[link=https://workspaces.openshift.com#https://github.com/eclipse/che-docs]
```

This is relevant for the second file mentioned above.
The che-docs repository features an link:https://github.com/eclipse-che/che-docs/blob/main/.vscode/extensions.json[extensions.json] file listing Visual Studio Code extensions
you need for writing Eclipse Che content.
When you launch the che-docs workspace for the first time,
these extensions are installed automatically:

image::/assets/img/contributing-to-eclipse-che-docs/extensions.png["Overview of automatically installed extensions, including the Vale extension."]

You can start working immediately,
with minimal additional setup required.

== Getting started

By launching the che-docs workspace, you're almost ready to go:

* You have a clone of the che-docs repository
* You have an IDE to write in.
* You have required extensions installed.
* You have a preview build script.

There are only three things left to do.

. Read the link:https://github.com/eclipse-che/che-docs/blob/main/CONTRIBUTING.adoc[contribution guidelines].
. Read how to link:https://redhat-documentation.github.io/supplementary-style-guide/#technical-examples[document commands] for Eclipse Che documentation.

> . Update the grammar- and spelling-checking link:https://marketplace.visualstudio.com/items?itemName=ChrisChinchilla.vale-vscode[Vale extension] by running the following command in the terminal:

Reviewer: ```suggestion
. Update the grammar- and spelling-checking link:https://marketplace.visualstudio.com/items?itemName=ChrisChinchilla.vale-vscode[Vale extension] by running the `vale-sync` command defined in the link:https://github.com/eclipse-che/che-docs/blob/main/devfile.yaml[devfile] or directly execute in the terminal:
```

> . Update the grammar- and spelling-checking link:https://marketplace.visualstudio.com/items?itemName=ChrisChinchilla.vale-vscode[Vale extension] by running the `vale-sync` command defined in the link:https://github.com/eclipse-che/che-docs/blob/main/devfile.yaml[devfile] or directly execute in the terminal:

Reviewer: ```suggestion
. Update the grammar- and spelling-checking link:https://marketplace.visualstudio.com/items?itemName=ChrisChinchilla.vale-vscode[Vale extension] by running the `vale-sync` command defined in the link:https://github.com/eclipse-che/che-docs/blob/main/devfile.yaml[devfile] or directly execute it in the terminal:
```
